src/GitHub.App/Services/GitHubContextService.cs

```diff
@@ -0,0 +1,474 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Windows;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Text.RegularExpressions;
+using System.Runtime.InteropServices;
+using GitHub.Services;
+using GitHub.Extensions;
+using GitHub.Primitives;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TextManager.Interop;
+using LibGit2Sharp;
+using Task = System.Threading.Tasks.Task;
+
+namespace GitHub.App.Services
+{
+    [Export(typeof(IGitHubContextService))]
+    public class GitHubContextService : IGitHubContextService
```

Comment:
We now have `GitHub.Services.Vssdk` - seeing as this service mostly deals with the VSSDK, would that be a better place for it?

Reply:
Given its dependencies on the `Microsoft.VisualStudio.*` assemblies, I think this makes sense.